test/observer.test.tsx

```diff
@@ -416,6 +416,64 @@ function runTestSuite(mode: "observer" | "useObserver") {
             expect(container.querySelector("span")!.innerHTML).toBe("1")
         })
     })
+
+    describe("error handling", () => {
+        test("errors should propagate", () => {
+            const x = mobx.observable.box(1)
+            const errorsSeen: any[] = []
+
+            class ErrorBoundary extends React.Component {
+                public static getDerivedStateFromError() {
+                    return { hasError: true }
+                }
+
+                public state = {
+                    hasError: false
+                }
+
+                public componentDidCatch(error: any, info: any) {
+                    errorsSeen.push("" + error)
+                }
+
+                public render() {
+                    if (this.state.hasError) {
+                        return <span>Saw error!</span>
+                    }
+                    return this.props.children
+                }
+            }
+
+            const C = obsComponent(() => {
+                if (x.get() === 42) {
+                    throw new Error("The meaning of life!")
+                }
+                return <span>{x.get()}</span>
+            })
+
+            // tslint:disable-next-line
+            const origErrorLogger = console.error
```

Comment:
Please replace with mockConsole, used eg. in useDisposable test

Reply:
I am wondering though, I tried running it locally without suppressing and why it would be showing it in a console it was caught? I wasn't fiddling with error boundaries too much yet but feels strange.

Reply:
It *is* really weird indeed, see also: https://github.com/facebook/react/issues/10474

The fun thing, the error doesn't escape any more (without the `<ErrorBoundary>` component you would correctly get an exception on `x.set(42)`), yet, it is _still_ being logged. 

I think it is meant as additional help to DEVs to see errors early. (After all error boundaries are for _unexpected_ exceptions, not normal errors). But in case of a test runner it is pretty annoying behavior.

Reply:
Updated with `mockConsole`. Neat.